File: src/codes/june_challenge/day6/queueReconstruction.js
/**
 * @param {number[][]} people
 * @return {number[][]}
 */
var reconstructQueue = function(people) {
    let arr = [];
    people.sort((a,b) => {
        if (a[0] == b[0]) {
            return a[1] - b[1]
        }
        return b[0] - a[0];
    });
    
    for (let i = 0; i < people.length; i++) {
       arr.splice(people[i][1], 0, people[i]);
    }
    return arr;
};

File: src/codes/june_challenge/day6/QueueReconstruction.java
package codes.june_challenge.day6;

import codes.fastio_java.InputReader;
import codes.fastio_java.OutputWriter;

import java.util.*;

public class QueueReconstruction {
    private static Integer apply(int[] e) {
        return e[0];
    }

    private static Integer apply2(int[] e) {
        return e[1];
    }

    public void solve(int testNumber, InputReader in, OutputWriter out) {
    }
    public int[][] reconstructQueue(int[][] people) {
        Arrays.sort(people, Comparator.comparing(QueueReconstruction::apply).reversed().thenComparing(QueueReconstruction::apply2));
        List<int[]> answer = new ArrayList<>();
        for (int[] person : people) {
            answer.add(person[1], person);
        }
        return answer.toArray(people);
    }
}
